pkg/commands/loading_commits.go

```diff
@@ -315,13 +315,39 @@ func (c *CommitListBuilder) setCommitMergedStatuses(refName string, commits []*m
 	return commits, nil
 }
 
+// Check if [branch "master"] or [branch "main"] has merge in .git/config. If there are no
+// merge paths available in the local config, use the init.defaultBranch from the global
+// configuration file. Ignoring file not found errors and default to baseBranch = "master".
+func getBaseBranch(c *CommitListBuilder) (string, error) {
+	baseBranch := ""
+	merge, _ := c.GitCommand.getLocalGitConfig("branch.master.merge")
+	merge, _ = c.GitCommand.getLocalGitConfig("branch.main.merge")
```

Comment:
Two questions on this part:
1) do we need to check globally here too or is this always going to be specific to the repo?
2) do we always want to use the value for branch.main.merge over branch.master.merge? I can imagine situations where a branch was coincidentally named 'main' before this all started (admittedly an edge case). Perhaps we should first see what the default is and then see if there's a merge value in the config.

Reply:
Thanks for the input.

1. To be honest, I'm not compelely sure if this config is used globally. From the brief research I did I noticed that the branch.*.merge seemed to only be in the local config. I may easily be wrong though.

2. This is an interesting point. I'm not sure if I understand you correctly -- Do you mean that branch.master.merge and branch.main.merge should be reordered so we would default to master as the base if both are in the file? Would this fix the edge case?

Reply:
After some consideration I thought that removing this first check would be the best solution here. I think at the moment this provides a way to color the list of commits correctly.

Of course when someone figures out a good way to reliably get the default remote branch, it can always be added later.

Please let me know if you disagree!

Reply:
I still think this might not quite be right: as @PGScully says the config was introduced to be read when creating new repos as opposed to dealing with an existing repo. It's possible that some people are setting it in local git configs as a conventional thing but we should find out if that's actually happening or not.

I don't think the global config should be considered because e.g. if I decide I want all future repos I create to have a default branch of 'blah', I wouldn't expect lazygit to treat all the repos on my computer as if 'blah' was their default branch.

I've done some googling on how to determine the default branch but I've had no luck. So I'm not too sure what the best path forward is other than more googling to work out what people are doing in the wild

Reply:
@jesseduffield is correct.  My old repositories still have `master` as the "main" branch, and my new ones have `main`.  Setting init.defaultBranch does not change the existing repository, only new ones (the clue is that it starts with init, for the `git init` command).

Reply:
I agree with both of you that this is not ideal solution. At the moment you could use the local config to overwrite the global config. The user can set the default branch in local config on the old repos to master and in new repos main will be automatically the default repo.